tensorflow_serving/batching/batching_session.cc

```diff
@@ -102,6 +105,127 @@ TensorSignature TensorSignatureFromSignatureDefs(
+    }
+};
+
+template<typename T>
+class PadTensor<T, 0> {
+ public:
+    Tensor operator()(Tensor input,
+                      Eigen::array<std::pair<int32, int32>, 0>  padding) {
+        TensorShape output_shape;
+        Tensor output(input.dtype(), output_shape);
+        typename TTypes<T, 0>::Tensor inputs = input.tensor<T, 0>();
+        output.tensor<T, 0>() = inputs;
+        return output;
+    }
+};
+
+template<typename T>
+Tensor PadTensorOfSpecificType(string tensor_name, Tensor tensor,
+                               std::map<string,
+                               std::vector<int>> max_dim_sizes) {
+    int num_dims = tensor.dims();
+    Tensor new_tensor;
+    switch (num_dims) {
+        case 0: {
+            Eigen::array<std::pair<int32, int32>, 0> padding;
+            padding = CreatePadding<0>(tensor, max_dim_sizes[tensor_name]);
+            PadTensor<T, 0> padding_functor = PadTensor<T, 0>();
+              new_tensor = padding_functor(tensor, padding);
+            break;
+        }
+        case 1: {
+            Eigen::array<std::pair<int32, int32>, 1> padding;
+            padding = CreatePadding<1>(tensor, max_dim_sizes[tensor_name]);
+            PadTensor<T, 1> padding_functor = PadTensor<T, 1>();
+            new_tensor = padding_functor(tensor, padding);
+            break;
+        }
+        case 2: {
+            Eigen::array<std::pair<int32, int32>, 2> padding;
+            padding = CreatePadding<2>(tensor, max_dim_sizes[tensor_name]);
+            PadTensor<T, 2> padding_functor = PadTensor<T, 2>();
+            new_tensor = padding_functor(tensor, padding);
+            break;
+        }
+        case 3: {
+            Eigen::array<std::pair<int32, int32>, 3> padding;
+            padding = CreatePadding<3>(tensor, max_dim_sizes[tensor_name]);
+            PadTensor<T, 3> padding_functor = PadTensor<T, 3>();
+            new_tensor = padding_functor(tensor, padding);
+            break;
+        }
+        case 4: {
+            Eigen::array<std::pair<int32, int32>, 4> padding;
+            padding = CreatePadding<4>(tensor, max_dim_sizes[tensor_name]);
+            PadTensor<T, 4> padding_functor = PadTensor<T, 4>();
+            new_tensor = padding_functor(tensor, padding);
+            break;
+        }
+        case 5: {
+            Eigen::array<std::pair<int32, int32>, 5> padding;
+            padding = CreatePadding<5>(tensor, max_dim_sizes[tensor_name]);
+            PadTensor<T, 5> padding_functor = PadTensor<T, 5>();
+            new_tensor = padding_functor(tensor, padding);
+            break;
+        }
+        case 6: {
+            Eigen::array<std::pair<int32, int32>, 6> padding;
+            padding = CreatePadding<6>(tensor, max_dim_sizes[tensor_name]);
+            PadTensor<T, 6> padding_functor = PadTensor<T, 6>();
+            new_tensor = padding_functor(tensor, padding);
+            break;
+        }
+        default:
+        // only ranks from 0 to 6 are supported
```

Comment:
Please arrange to return an error then, instead of just not doing the padding.

Also, please document this limitation in the header file.